core/templates/dev/head/dashboard/Dashboard.js

```diff
@@ -34,11 +36,19 @@ oppia.controller('Dashboard', [
     };
 
     $rootScope.loadingMessage = 'Loading';
-    $http.get('/dashboardhandler/data').then(function(response) {
-      var data = response.data;
-      $scope.explorationsList = data.explorations_list;
-      $scope.collectionsList = data.collections_list;
-      $rootScope.loadingMessage = '';
-    });
+    DashboardBackendApiService.fetchDashboardData().then(
+      function(response) {
+        $scope.explorationsList = response.explorations_list;
+        $scope.collectionsList = response.collections_list;
+        $scope.dashboardStats = response.dashboard_stats;
+        $rootScope.loadingMessage = '';
+      },
+      function(errorStatus) {
+        var ERROR_STATUSES_FOR_WARNING = [400, 401, 404, 500];
```

Comment:
Also, I think this name isn't very descriptive. Could you call it FATAL_ERROR_CODES or something like that?